db/commitlog/commitlog.cc

```diff
@@ -96,13 +96,19 @@ class db::cf_holder {
     virtual void release_cf_count(const cf_id_type&) = 0;
 };
 
+static size_t adjusted_segment_size_in_mb(size_t s) {
+    return std::min<size_t>(std::numeric_limits<db::position_type>::max() / (1024 * 1024), std::max<size_t>(s, 1));
```

Comment:
why not take the minimum into account here?

e.g.
```c++
static size_t adjusted_segment_size_in_mb(size_t s, size_t min_size, size_t max_size = ) {
    return std::min<size_t>(std::numeric_limits<db::position_type>::max() / (1024 * 1024), std::max<size_t>(s, min_size));
}
```

Reply:
We don't really have a minimum segment size. At least not defined.

Reply:
Well, 1 seems to be the minimum.
I had in mind `min_size = 2 * smp::count * c.commitlog_segment_size_in_mb` below, but I see now that it relates to the `commitlog_total_space_in_mb`, not to `commitlog_segment_size_in_mb`.

For symmetry reasons, why not have an `adjusted_total_space_in_mb` function too?